app/lib/account/backend.dart

```diff
@@ -199,26 +200,62 @@ class AccountBackend {
     return likes.map((Like l) => l.package).toList();
   }
 
-  /// Creates an a package like entry for the given [user] and [package].
+  /// Creates and returns a package like entry for the given [user] and
+  /// [package], and increments the 'likes' property on [package].
   Future<Like> likePackage(User user, String package) async {
-    final newLike = Like()
-      ..parentKey = user.key
-      ..id = package
-      ..created = DateTime.now().toUtc();
-    await _db.commit(inserts: [newLike]);
-    return newLike;
+    final like = await _db.withTransaction<Like>((tx) async {
+      final packageKey = _db.emptyKey.append(Package, id: package);
+      final p = await tx.lookupValue<Package>(packageKey);
+      if (p == null) {
+        throw NotFoundException.resource(package);
+      }
+
+      final key =
+          _db.emptyKey.append(User, id: user.id).append(Like, id: package);
+      final oldLike = await tx.lookupValue<Like>(key, orElse: () => null);
+
+      if (oldLike != null) {
+        tx.rollback();
+        return oldLike;
+      }
+
+      p.likes++;
+      final newLike = Like()
+        ..parentKey = user.key
+        ..id = p.id
+        ..created = DateTime.now().toUtc();
+
+      tx.queueMutations(inserts: [p, newLike]);
+      tx.commit();
+      return newLike;
+    });
+
+    return like;
   }
 
-  /// Delete a package like entry for the given [user] and [package] if it exists.
+  /// Delete a package like entry for the given [user] and [package] if it
+  /// exists, and decrements the 'likes' property on [package].
   Future<void> unlikePackage(User user, String package) async {
-    final key =
-        _db.emptyKey.append(User, id: user.id).append(Like, id: package);
-    try {
-      await _db.lookupValue(key);
-    } on KeyNotFoundException {
-      return;
-    }
-    await _db.commit(deletes: [key]);
+    await _db.withTransaction<void>((tx) async {
+      final packageKey = _db.emptyKey.append(Package, id: package);
+      final p = await tx.lookupValue<Package>(packageKey);
```

Comment:
```suggestion
      final p = await tx.lookupValue<Package>(packageKey, orElse: () => null);
```
Just for sanity maybe do a `grep lookupValue` and see if you missed `orElse` somewhere :)